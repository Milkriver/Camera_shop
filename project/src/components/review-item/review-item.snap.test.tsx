import {render} from '@testing-library/react';
import { createMemoryHistory } from 'history';
import HistoryRouter from '../history-router/history-router';
import ReviewItem from './review-item';

const history = createMemoryHistory();
const mockReview = {
        id: '',
        userName: '',
        advantage: '',
        disadvantage: '',
        review: '',
        rating: 1,
        createAt: '',
        cameraId: 1,
};


describe('Component: ReviewItem', () => {
  it('should render correctly', () => {
    const {container} = render(
        <HistoryRouter history={history}>
            <ReviewItem review={mockReview}/>
        </HistoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});