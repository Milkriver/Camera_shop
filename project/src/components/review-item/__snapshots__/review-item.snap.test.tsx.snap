// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ReviewItem should render correctly 1`] = `
<div>
  <li
    class="review-card"
  >
    <div
      class="review-card__head"
    >
      <p
        class="title title--h4"
      />
      <time
        class="review-card__data"
        datetime=""
      >
        NaN undefined
      </time>
    </div>
    <div
      class="rate review-card__rate"
    >
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-full-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-star"
        />
      </svg>
      <p
        class="visually-hidden"
      >
        Оценка: 
        1
      </p>
    </div>
    <ul
      class="review-card__list"
    >
      <li
        class="item-list"
      >
        <span
          class="item-list__title"
        >
          Достоинства:
        </span>
        <p
          class="item-list__text"
        />
      </li>
      <li
        class="item-list"
      >
        <span
          class="item-list__title"
        >
          Недостатки:
        </span>
        <p
          class="item-list__text"
        />
      </li>
      <li
        class="item-list"
      >
        <span
          class="item-list__title"
        >
          Комментарий:
        </span>
        <p
          class="item-list__text"
        />
      </li>
    </ul>
  </li>
</div>
`;
